content/src/storage/FileSystemContentStorage.ts

```diff
@@ -76,19 +72,87 @@ export class FileSystemContentStorage implements ContentStorage {
     return new Map(checks)
   }
 
+  async store(id: string, content: Uint8Array | Readable): Promise<void> {
+    if (content instanceof Uint8Array || Buffer.isBuffer(content)) {
+      return fs.promises.writeFile(this.getFilePath(id), content)
+    } else {
+      return pipe(content, fs.createWriteStream(this.getFilePath(id)))
+    }
+  }
+
+  async size(id: string): Promise<number | undefined> {
+    return (await this.stats(id))?.size
+  }
+
+  async fixContentItem(id: string, oldContentPath: string): Promise<void> {
+    return fs.promises.rename(path.join(oldContentPath, id), this.getFilePath(id))
+  }
+
+  create(id: string): UnsavedContentItem {
+    // Check if it doesn't exist
+    return createUnsavedContentItem(id, this.getTmpFilePath(id), this.getFilePath(id))
+  }
```

Comment:
I'd move this to a helper outside the storage component. How will this work for S3 or IPFS? I think tmp file management should be separated from the storage. And adding a storage.storeFromFile(tmpPath) method to handle the copy/upload/renaming